safe_transaction_service/history/services/transaction_service.py

```diff
@@ -371,3 +374,585 @@ def serialize_all_txs_v2(
+                    WHERE erc20.ethereum_tx_id = et.tx_hash
+                )
+                AND NOT EXISTS (
+                    SELECT 1 FROM history_erc721transfer erc721
+                    WHERE erc721.ethereum_tx_id = et.tx_hash
+                )
+
+                UNION ALL
+
+                -- Module transactions
+                SELECT itx.timestamp as execution_date, itx.ethereum_tx_id as transaction_hash
+                FROM history_moduletransaction modtx
+                JOIN history_internaltx itx ON modtx.internal_tx_id = itx.id
+                WHERE modtx.safe = %s
+
+                UNION ALL
+
+                -- Standalone ERC20 transfers
+                SELECT erc20.timestamp as execution_date, erc20.ethereum_tx_id as transaction_hash
+                FROM history_erc20transfer erc20
+                WHERE (erc20.to = %s OR erc20._from = %s)
+                AND NOT EXISTS (
+                    SELECT 1 FROM history_multisigtransaction mt
+                    WHERE mt.ethereum_tx_id = erc20.ethereum_tx_id
+                )
+                AND NOT EXISTS (
+                    SELECT 1 FROM history_moduletransaction modtx
+                    JOIN history_internaltx itx ON modtx.internal_tx_id = itx.id
+                    WHERE itx.ethereum_tx_id = erc20.ethereum_tx_id
+                )
+
+                UNION ALL
+
+                -- Standalone ERC721 transfers
+                SELECT erc721.timestamp as execution_date, erc721.ethereum_tx_id as transaction_hash
+                FROM history_erc721transfer erc721
+                WHERE (erc721.to = %s OR erc721._from = %s)
+                AND NOT EXISTS (
+                    SELECT 1 FROM history_multisigtransaction mt
+                    WHERE mt.ethereum_tx_id = erc721.ethereum_tx_id
+                )
+                AND NOT EXISTS (
+                    SELECT 1 FROM history_moduletransaction modtx
+                    JOIN history_internaltx itx ON modtx.internal_tx_id = itx.id
+                    WHERE itx.ethereum_tx_id = erc721.ethereum_tx_id
+                )
+
+                UNION ALL
+
+                -- Standalone Ether transfers
+                SELECT itx.timestamp as execution_date, itx.ethereum_tx_id as transaction_hash
+                FROM history_internaltx itx
+                WHERE (itx.to = %s OR itx._from = %s)
+                AND itx.call_type = 0 AND itx.value > 0
+                AND NOT EXISTS (
+                    SELECT 1 FROM history_multisigtransaction mt
+                    WHERE mt.ethereum_tx_id = itx.ethereum_tx_id
+                )
+                AND NOT EXISTS (
+                    SELECT 1 FROM history_moduletransaction modtx
+                    JOIN history_internaltx itx2 ON modtx.internal_tx_id = itx2.id
+                    WHERE itx2.ethereum_tx_id = itx.ethereum_tx_id
+                )
+            ) combined
+        )
+        SELECT COUNT(DISTINCT (execution_date, transaction_hash))
```

Comment:
@Uxio0 I think could give as inconsistent count, same transaction hash in the same date could contains several ERC20 transfers, would makes sense use trace_address and log_index to differenciate it.